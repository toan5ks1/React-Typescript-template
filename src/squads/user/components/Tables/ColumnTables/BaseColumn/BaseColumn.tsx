import TypographyBase, { TypographyBaseProps } from "src/components/Typographys/TypographyBase";

export interface BaseColumnProps extends Omit<TypographyBaseProps, "variant"> {
    content?: string | null;
}

function BaseColumn({ content, ...rest }: BaseColumnProps) {
    return (
        <TypographyBase variant="body2" {...rest}>
            {content}
        </TypographyBase>
    );
}
export default BaseColumn;
