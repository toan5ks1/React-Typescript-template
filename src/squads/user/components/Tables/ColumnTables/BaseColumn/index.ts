export * from "./BaseColumn";
export { default } from "./BaseColumn";
