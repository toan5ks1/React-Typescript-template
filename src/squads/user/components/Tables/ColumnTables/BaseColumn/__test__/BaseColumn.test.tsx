import BaseColumn, { BaseColumnProps } from "../BaseColumn";

import { render, screen } from "@testing-library/react";

describe("BaseColumn/>", () => {
    const renderComponent = (props?: Partial<BaseColumnProps>) => {
        return render(<BaseColumn content={"Test"} {...props} />);
    };

    it("should match snapshot", () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });

    it("should render content when having value", () => {
        renderComponent();

        expect(screen.getByText("Test")).toBeInTheDocument();
        expect(screen.getByText("Test")).toHaveClass("MuiTypography-body2");
    });
});
