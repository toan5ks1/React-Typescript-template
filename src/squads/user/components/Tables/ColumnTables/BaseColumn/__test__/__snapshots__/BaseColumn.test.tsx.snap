// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseColumn/> should match snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
}

<div>
  <p
    class="MuiTypography-root MuiTypography-body2 emotion-0"
  >
    Test
  </p>
</div>
`;
